mmdet3d/datasets/kitti_dataset.py

```diff
@@ -670,22 +674,61 @@ def convert_valid_bboxes(self, box_dict, info):
                 sample_idx=sample_idx,
             )
 
-    def show(self, results, out_dir, show=True):
+    def _build_default_pipeline(self):
+        """Build the default pipeline for this dataset."""
+        if self.modality['use_camera']:
+            pipeline = [
+                dict(
+                    type='LoadPointsFromFile',
+                    coord_type='LIDAR',
+                    load_dim=4,
+                    use_dim=4,
+                    file_client_args=dict(backend='disk')),
+                dict(type='LoadImageFromFile'),
+                dict(
+                    type='DefaultFormatBundle3D',
+                    class_names=self.CLASSES,
+                    with_label=False),
+                dict(type='Collect3D', keys=['points', 'img'])
+            ]
+        else:
+            pipeline = [
+                dict(
+                    type='LoadPointsFromFile',
+                    coord_type='LIDAR',
+                    load_dim=4,
+                    use_dim=4,
+                    file_client_args=dict(backend='disk')),
+                dict(
+                    type='DefaultFormatBundle3D',
+                    class_names=self.CLASSES,
+                    with_label=False),
+                dict(type='Collect3D', keys=['points'])
+            ]
```

Comment:
Simplify these two cases? First define the pipeline with a list including the shared 3 pipelines, then insert dict(type='LoadImageFromFile') for different cases.

Reply:
Good idea.